core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -160,6 +173,20 @@ DeleteFile[] forDataFile(long sequenceNumber, DataFile file) {
         .toArray(DeleteFile[]::new);
   }
 
+  private DeleteFile[] limitWithoutColumnStatsFiltering(
+      long sequenceNumber, DeleteFileGroup partitionDeletes) {
+
+    if (partitionDeletes == null) {
+      return globalDeletes.filter(sequenceNumber);
+    } else if (globalDeletes == null) {
+      return partitionDeletes.filter(sequenceNumber);
+    } else {
+      DeleteFile[] matchingGlobalDeletes = globalDeletes.filter(sequenceNumber);
+      DeleteFile[] matchingPartitionDeletes = partitionDeletes.filter(sequenceNumber);
+      return ObjectArrays.concat(matchingGlobalDeletes, matchingPartitionDeletes, DeleteFile.class);
```

Comment:
Here we can not do further filtering by the data file column stats like here: https://github.com/apache/iceberg/pull/8278/files#diff-237e3fe026d656e4410a14e1be56d5af1e38a49e5cda651da743bc5a3e1c6bcfL158. This may increase unnecessary delete files matched to the data file.

Reply:
@aokolnychyi I think we should do some benchmarks about the performance of the table scan while disabling the table property.

Reply:
I am not saying the stats filtering should be disabled by default or that it does not make sense in general. In some cases, it will help. In some cases, it will harm. I want to have a flag to disable column stats filtering and only rely on partition pruning and sequence numbers as that step is expensive and may be useless. It may be highly beneficial for position deletes generated by Spark and in some other cases when lower/upper boundaries are random.

Reply:
Yeah, I see. It makes sense to keep an option here for some cases.